src/Framework/Framework/Binding/ControlPropertyBindingDataContextChangeAttribute.cs

```diff
@@ -29,11 +31,26 @@ public ControlPropertyBindingDataContextChangeAttribute(string propertyName, int
                 throw new Exception($"The property '{PropertyName}' was not found on control '{control.Metadata.Type}'!");
             }
 
+            [DoesNotReturn]
+            void ThrowDataContextMismatch(IAbstractBinding binding)
+            {
+                var sb = new StringBuilder();
+                sb.Append($"The '{controlProperty.Name}' property contains invalid data-binding. ");
+                sb.Append($"Binding '{binding}' could not be constructed on current context ({dataContext})");
+                throw new Exception(sb.ToString());
```

Comment:
On ResolvedBinding, there is a `Errors.GetErrorMessage` method which could be useful, but we'd have to expose it on the IAbstractBinding interface